soci/ztoc.go

```diff
@@ -54,30 +54,32 @@ type FileMetadata struct {
 }
 
 type Ztoc struct {
-	Version             string
-	BuildToolIdentifier string
-
-	Metadata []FileMetadata
+	Version                 string
+	BuildToolIdentifier     string
+	CompressedArchiveSize   FileSize
+	UncompressedArchiveSize FileSize
+	TOC                     TOC
+	CompressionInfo         CompressionInfo
+}
 
-	CompressedFileSize   FileSize
-	UncompressedFileSize FileSize
-	MaxSpanID            SpanID //The total number of spans in Ztoc - 1
-	ZtocInfo             ztocInfo
-	IndexByteData        []byte
+type CompressionInfo struct {
```

Comment:
Should compression info contain the algorithm?

Reply:
It didn't make any sense to add it just yet, since we don't have anything but gzip here (fbs has it already and this is the final destination for serialization). Before putting algorithm in the compression info in Ztoc struct, it's better to devise similarities and differences between Ztocs for gzip and zstd and model how will the new reality look like.